### ISSUE TITLE
random_integer_uniform clarity

### ISSUE THREAD

**Issue by chris-wood** [cfrg/draft-irtf-cfrg-blind-signatures#155]
Section 3.
random_integer_uniform: The use of boundaries M <= R < N confused me a
little. Not thoroughly reading this section and then generating r =
random_integer_uniform(1, n) I can see that someone might implement this
as  M <= R <= N. But maybe that's just me.

Section 3
 - In the description of random_integer_uniform(M, N), cryptographic security isn't mentioned, but is mentioned for random(n). Should this mention that the random source needs to be cryptographically secure?

**Comment by jedisct1** [comment]
This is matching how functions to generate random numbers in a range work in most languages.

But maybe we should explain the boundaries in plain English instead.

And mentioning that it should be cryptographically secure wouldn't be a bad thing either.

**Comment by chris-wood** [comment]
@jedisct1 it's not clear to me what we would change in this function description. We already say that the output should be uniformly distributed, so what would "cryptographically secure" look like here?

**Comment by jedisct1** [comment]
It must be unpredictable to an adversary as well.

**Comment by chris-wood** [comment]
To me that's implied by the output being uniformly distributed. Can you please suggest concrete text?

**Comment by chris-wood** [comment]
But it doesn't really need to be secret. In fact, the random bytes generated from this function can be prepended to messages that are signed and made public! I don't think that's quite the right phrasing. (I don't know right now what is the best phrasing.)

**Comment by jedisct1** [comment]
Oh, right!

There's probably no need to change anything to the current phrasing, then.
